pkg/cmd/pulumi/new.go

```diff
@@ -390,6 +451,205 @@ func useSpecifiedDir(dir string) (string, error) {
 	return cwd, nil
 }
 
+type AIPromptRequestBody struct {
+	Language       string `json:"language"`
+	Instructions   string `json:"instructions"`
+	Model          string `json:"model"`
+	ResponseMode   string `json:"responseMode"`
+	ConversationID string `json:"conversationId"`
+	ConnectionID   string `json:"connectionId"`
+}
+
+// Iteratively prompt the user for input, sending their input as a prompt tp Pulumi AI
+// Stream the response back to the console, and repeat until the user is done.
+func runAINew(
+	ctx context.Context,
+	args newArgs,
+	opts display.Options,
+	userName string,
+) (conversationURL string, err error) {
+	languageOptions := []string{
+		"TypeScript",
+		"JavaScript",
+		"Python",
+		"Go",
+		"C#",
```

Comment:
Is there a reason we don't offer YAML and Java options here?  I think we should (even if they are less used).

Reply:
We didn't have YAML because it wasn't yet supported in the AI service for project rendering. I've added that support here:
https://github.com/pulumi/pulumi.ai/pull/508
and have updated this PR to support both. (Java was just a simple omission)